app/assets/javascripts/discourse/app/models/topic.js

```diff
@@ -487,8 +487,10 @@ const Topic = RestModel.extend({
               (group) => group.name === this.category?.reviewable_by_group_name
             ) &&
             !(
-              this.siteSettings.tl4_delete_posts_and_topics &&
-              deleted_by.trust_level >= 4
+              this.siteSettings.delete_all_posts_and_topics_allowed_groups &&
+              deleted_by.isInAnyGroups(
+                this.siteSettings.delete_all_posts_and_topics_allowed_groups
+              )
```

Comment:
What this front-end code does is redirect you after deleting a topic if you don't have access to see it after it's deleted.

Reply:
> Okwenziwa yile khodi yangaphambili ikuqondisa kabusha ngemva kokususa isihloko uma ungakwazi ukufinyelela ukusibona ngemva kokuthi sisusiwe.

``` suggestion
              this.siteSettings.delete_all_posts_and_topics_allowed_groups &&
              deleted_by.isInAnyGroups(
                this.siteSettings.delete_all_posts_and_topics_allowed_groups
              )https://github.com/discourse/discourse/pull/24866#discussion_r1425207163
```